### DISCUSSION TITLE
Disabling core resource types and "installing" 3rd party resource types

### DISCUSSION THREAD

**Discussion by mvdkleijn** [concourse/concourse#6267] [Help & Support] [answered]
Hi,

I'm wondering if there is a way to either one or both of the following things:

1. I'd like to be able to disable core resource types. For example, I'd like to remove the availability of the S3 resource type by listing it as disabled in the Concourse config.

2. I'd like to be able to "install" or "import" 3rd party resource types, so they are available on every worker similar to the core resource types. I.e. when starting, the worker would pull the resource type and in the pipeline yaml, people would not need to add an entry under `resource_types`

Possible?

**Comment by chenbh** [comment, marked as answer]
Yes there is! Every distribution of Concourse comes with a `resource-types` directory, and the worker will scan this to determine what base resource-types to advertise to the atc. On the docker image this is located under `/usr/local/concourse/resource-types/`

To add a base resource type, you'll have to create a directory and populate it with:
- `rootfs.tgz` contains the filesystem of the resource, the easiest way to get this would be something like 
  `docker export $(docker create concourse/registry-image-resource) | gzip > rootfs.tgz`
- `resource_metadata.json` which contains metadata about the resource, the one for registry-image-resource looks like:
```json
{
  "type": "registry-image",
  "version": "0.12.0",
  "privileged": false,
  "unique_version_history": false
}
```

Once that's done and you've started the worker, you can verify what workers are advertising what resource types by going to `<concourse url>/api/v1/workers`


To remove you can just delete the directory and start the worker. Once the resource type is removed, any pipelines that use it will get an error something along the lines of "no workers matching resources: s3". Note that this doesn't stop anyone from declaring and using it as a resource-type in the pipeline itself.

**Reply by mvdkleijn** [reply]
Thanks a ton! I couldn't find any docs on this at all...

**Reply by taylorsilva** [reply]
There are some docs for this now for anyone looking: https://concourse-ci.org/concourse-worker.html#installing-or-upgrading-bundled-resource-types
